src/Symfony/Component/Ldap/LdapClient.php

```diff
@@ -79,67 +49,37 @@ public function bind($dn = null, $password = null)
      */
     public function find($dn, $query, $filter = '*')
     {
-        if (!is_array($filter)) {
-            $filter = array($filter);
-        }
+        @trigger_error('The "find" method is deprecated since version 3.1 and will be removed in 4.0. Use the "query" method instead.', E_USER_DEPRECATED);
```

Comment:
Do we really need to deprecate method explicitly as the whole class is already deprecated?